docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+The `magnet_uri_hash` field MUST contain the KECCAK-256 hash of the magnet link created from the `WakuMessageArchive`.
+
+The `magnet_uri` field MUST contain the magnet link created from the `WakuMessageArchive`.
+
+```
+syntax = "proto3"
+
+message WakuMessageArchiveMetadata {
+  uint64 from = 1
+  uint64 to = 2
+  string contentTopic = 3
+  string magnet_uri_hash = 4
+  string magnet_uri = 5
+}
+
+message WakuMessageArchiveIndex {
+  map<string, WakuMessageArchiveMetadata> archives = 1
+}
+```
+
+The community owner node MUST create a `WakuMessageArchiveIndex` every time it creates a new `WakuMessageArchive`.
+
+For every created `WakuMessageArchive`, there MUST be a `WakuMessageArchiveMetadata` entry in the index map.
+
+The the community owner node MUST derive a magnet link from the newly created `WakuMessageArchiveIndex` so it can be distributed to community member nodes.
+
+## Message archive distribution
+
+Message archives are available via the BitTorrent network as soon as magnet links for them have been created.
+Other community member nodes will download the message archives from the BitTorrent network once they receive a magnet link that contains a message archive index.
+
+The community owner node MUST send magnet links containing message archive indices to a special community channel. The topic of that special channel consists of the community id followed by the string `-archives`:
+
+```
+{community_id}-archives
+```
+
+All messages sent with this topic MUST be instances of `ApplicationMetadataMessage` ([6/PAYLOADS](/specs/6-payloads)) with a `payload` of `CommunityMessageArchiveIndex`.
+
+Only the community owner has permission to send messages with this topic.
+Community members MUST NOT have permission to send messages with this topic.
+However, community member nodes MUST subscribe to this topic to receive message archive indices.
+
+## Canonical message histories
+
+Only community owners are allowed to distribute messages with magnet links via the magnet link channel. Community members MUST NOT be allowed to distribute magnet links. Since the magnet links are created from the community owner node's database (and previously distributed archives), the message history provided by the community owner becomes the canonical message history and single source of truth for the community.
+
+Community member nodes MUST replace messages in their local databases with the messages extracted from archives within the same time range. Messages that didn't receive the community owner node MUST be removed and are no longer part of the message history of interest, even if it already existed in a community member node's database.
+
+
+## Fetching message archives
+
+Generally, fetching message archives is a tree step process:
+
+1. Receive message archive index signal, download index, then determine which message archives to download
+3. Download individual archives
```

Comment:
That's 2 steps. Needs update.

Reply:
I don't think we want individual archives, I think we want all of a community's history to be in a single binary blob (with a separator string inserted into the binary blob at the point each new blob is prepended to the prexisting blob, to enable easy splitting of the blob for the purposes of enabling partial download of a community's history and/or keeping the storage used by a community with in a set bound (in phase 2 of this project - enabling partial downloads and fixed storage use per community isn't needed for the MVP)

Reply:
How do we prevent nodes from redownloading all the data they have downloaded already, if they don't know at this point what that `blob` looks like (they need to download it first, then can perform that splitting).

Using the archive index, I've tried to come up with a solution to account for that. It also provides flexibility for nodes to selectively decide for what date range they want to fetch archives (something the blob-only solution won't do, unless I'm missing something here)